### ISSUE TITLE
`3:10  error    'doesSomething' is already defined                                                  no-redeclare`

### ISSUE THREAD

**Issue by garyking** [typescript-eslint/typescript-eslint#2562]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JS
module.exports = {
  extends: [],
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  rules: {}
}
```

```TS
function doesSomething(item: string): string;

function doesSomething(item: number): number;

function doesSomething(item: string | number): string | number {
  return item;
}
```

**Expected Result**

No error.

**Actual Result**

Errors:

```
  3:10  error    'doesSomething' is already defined                                                  no-redeclare
  5:10  error    'doesSomething' is already defined                                                  no-redeclare
```

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `4.1.1` |
| `TypeScript`                | `4.0.2` |
| `ESLint`                    | `7.9.0` |
| `node`                      | `14.8.0` |

**Comment by bradzacher** [comment]
you need to update your config 

- [This is a listed action item for users in the v4.0.0 release notes](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0)
- [In case you missed the release note, this is listed in the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code)
- [Our extension rule's docs lists the config you should use](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-redeclare.md)
